File: persistencia/src/main/java/co/edu/uniquindio/proyecto/dto/TransaccionGetDTO.java
package co.edu.uniquindio.proyecto.dto;

import co.edu.uniquindio.proyecto.entidades.MetodoDePago;
import lombok.*;

import java.time.LocalDateTime;

@Setter
@Getter
@AllArgsConstructor
@NoArgsConstructor
@ToString
public class TransaccionGetDTO {

    private LocalDateTime fecha;
    private Double total;
    private String usuarioCompradorCedula;
    private String usuarioVendedorCedula;
    private MetodoDePago metodoDePago;
    private ProductoGetDTO producto;
    private Integer unidadesCompradas;


}

File: my-app/src/app/modelo/transaccion-get-dto.ts
import { ProductoGetDTO } from './producto-get-dto';

export class TransaccionGetDTO {
  fecha: Date = new Date();
  total: number = 0;
  usuarioCompradorCedula: string = '';
  usuarioVendedorCedula: string = '';
  metodoDePago: String = "";
  producto: ProductoGetDTO;
  unidadesCompradas: number = 0;

  constructor(
    fecha: Date,
    total: number ,
    usuarioCompradorCedula: string,
    usuarioVendedorCedula: string,
    metodoDePago: string,
    producto: ProductoGetDTO,
    unidadesCompradas: number,
  ) {
    this.fecha = fecha;
    this.total = total;
    this.usuarioCompradorCedula = usuarioCompradorCedula;
    this.usuarioVendedorCedula = usuarioVendedorCedula;
    this.metodoDePago = metodoDePago;
    this.producto = producto;
    this.unidadesCompradas = unidadesCompradas;
  }
}
